src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -42,6 +45,18 @@ public class BatchHandler : DatabaseRequestHandler
                 else
                     ThrowNotSupportedType(contentType);
 
+                for (int i = 0; i < command.ParsedCommands.Count; i++)
+                {
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Type != CommandType.PUT)
+                        continue;
+
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id?.EndsWith("/") != true)
+                        continue;
+
+                    command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id =
+                        await ServerStore.GenerateClusterIdentityAsync(command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id, Database.Name);
```

Comment:
What do you think about moving identity generation to `BatchRequestParser.ParseAsync` and calling this method once we encounter such a PUT command? We could avoid iterating over the parsed commands here.